java/food-packaging/src/main/java/org/acme/foodpackaging/domain/Job.java

```diff
@@ -40,15 +41,19 @@ public class Job {
     @NextElementShadowVariable(sourceVariableName = "jobs")
     private Job nextJob;
 
-    /** Start is after cleanup. */
-    @ShadowVariable(variableListenerClass = StartDateTimeUpdatingVariableListener.class, sourceVariableName = "line")
-    @ShadowVariable(variableListenerClass = StartDateTimeUpdatingVariableListener.class, sourceVariableName = "previousJob")
+    /**
+     * Start is after cleanup.
+     */
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "line")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "previousJob")
     private LocalDateTime startCleaningDateTime;
 
-    @PiggybackShadowVariable(shadowVariableName = "startCleaningDateTime")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "line")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "previousJob")
     private LocalDateTime startProductionDateTime;
 
-    @PiggybackShadowVariable(shadowVariableName = "startCleaningDateTime")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "line")
+    @CascadingUpdateShadowVariable(targetMethodName = "updateStartCleaningDateTime", sourceVariableName = "previousJob")
```

Comment:
Observation: that's a lot of boilerplate. The annotation - 6 times. The name of the method - 6 times. The name of each variable - three times.

Reply:
I now wonder if we should've put the annotation on the method instead. That would completely eliminate all this repetition.

Reply:
Done!